### PR TITLE
[frontendTable] Add Table.

### PR DESCRIPTION
**Pull request by Piwakk**

### CODE REVIEW

File: frontend/src/components/utils/table/TableHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export type Column = {
+    key: string;
+    header: any;
+    render?: (data: any) => any;
+    sorting?: Sorting;
+    onChangeSorting?: (newSorting?: Sorting) => void;
+};
+
+export type Columns = Column[];
```
Comment (abocquet) [inline frontend/src/components/utils/table/TableHeader.tsx:13]: Idk if simply ˋColumn[]` everywhere wouldn’t be simpler

File: frontend/src/components/utils/table/TableBody.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const Row = ({
+    columns,
+    data,
+    index
+}: {
+    columns: Columns;
+    data: object;
+    index: number;
+}) => (
+    <tr role="row" className={index % 2 == 0 ? "even" : "odd"}>
```
Comment (abocquet) [inline frontend/src/components/utils/table/TableBody.tsx:13]: In bootstrap there is a way to have colour alternance without adding specific classes to HTML

File: frontend/src/components/utils/table/sorting.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ */
+export const cycleSorting = (current?: Sorting) =>
+    current === Sorting.Unsorted
+        ? Sorting.Ascending
+        : current === Sorting.Ascending
+        ? Sorting.Descending
+        : current === Sorting.Descending
+        ? Sorting.Unsorted
+        : undefined;
+
```
Comment (abocquet) [inline frontend/src/components/utils/table/sorting.ts:30]: What happens if it’s undefined ?

File: frontend/src/components/utils/table/TableBody.tsx (hunk 2 above)
Comment (Piwakk) [reply frontend/src/components/utils/table/TableBody.tsx:13]: I don't know, I saw this on Tabler.io's website

File: frontend/src/components/utils/table/sorting.ts (hunk 3 above)
Comment (Piwakk) [reply frontend/src/components/utils/table/sorting.ts:30]: It just stays undefined?

Comment (Piwakk) [reply frontend/src/components/utils/table/sorting.ts:30]: Not sure I understood the question

File: frontend/src/components/utils/table/TableHeader.tsx (hunk 1 above)
Comment (Piwakk) [reply frontend/src/components/utils/table/TableHeader.tsx:13]: Maybe, I think it's not the most useful part of the code anyways … :D

**Pull request merged**